app/controllers/concerns/collection.rb

```diff
@@ -8,10 +8,17 @@ module Collection
 
   included do
     def resolve_collection
-      result = sort(search(policy_scope(resource)))
+      result = filter_by_tags(sort(search(policy_scope(resource))))
```

Comment:
Observation: filtering before sort.

Reply:
Ideally the ordering here doesn't really matter as ActiveRecord constructs the query at the end when the `result` is being used. However, somehow the `filter_by_tags` tends to give an array as a result and not a CollectionProxy :confused: which I don't really understand why, it might be connected with the fact that we're querying a jsonb field. So TL;DR if we swap the order here, the chaining will just not work :disappointed:

Reply:
Interesting. I wonder where is the hiccup.

Reply:
My guess is that the `jsonb` support in AR is relatively new and it might be just poorly implemented. Anyway, theoretically having these in different orders shouldn't alter the query, but I agree that it goes against human logic :see_no_evil: